src/Symfony/Component/PropertyInfo/Extractor/PhpDocExtractor.php

```diff
@@ -151,6 +151,77 @@ public function getTypes($class, $property, array $context = [])
         return [new Type(Type::BUILTIN_TYPE_ARRAY, false, null, true, new Type(Type::BUILTIN_TYPE_INT), $types[0])];
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function getTypesFromConstructor($class, $property)
+    {
+        $docBlock = $this->getDocBlockFromConstructor($class, $property);
+
+        if (!$docBlock) {
+            return;
+        }
+
+        $types = [];
+        /** @var DocBlock\Tags\Var_|DocBlock\Tags\Return_|DocBlock\Tags\Param $tag */
+        foreach ($docBlock->getTagsByName('param') as $tag) {
+            if ($tag && null !== $tag->getType()) {
+                $types = array_merge($types, $this->phpDocTypeHelper->getTypes($tag->getType()));
+            }
+        }
+
+        if (!isset($types[0])) {
+            return;
+        }
+
+        return $types;
+    }
+
+    /**
+     * Gets the DocBlock from a constructor.
+     *
+     * @param string $class
+     * @param string $property
+     *
+     * @return DocBlock|null
+     */
+    private function getDocBlockFromConstructor($class, $property)
+    {
+        try {
+            $reflectionClass = new \ReflectionClass($class);
+        } catch (\ReflectionException $e) {
+            return null;
+        }
+        $reflectionConstructor = $reflectionClass->getConstructor();
+        if (!$reflectionConstructor) {
+            return null;
+        }
+
+        try {
+            $docBlock = $this->docBlockFactory->create($reflectionConstructor, $this->contextFactory->createFromReflector($reflectionConstructor));
+
+            return $this->filterDocBlockParams($docBlock, $property);
+        } catch (\InvalidArgumentException $e) {
+            return null;
+        }
+    }
+
+    /**
+     * @param DocBlock $docBlock
+     * @param string   $allowedParam
+     *
+     * @return DocBlock
+     */
+    private function filterDocBlockParams(DocBlock $docBlock, $allowedParam)
```

Comment:
You should use type hints and return types wherever possible as this is for 4.2 and new code if I am right

Reply:
@OskarStark I applied the type hints in #30335